narwhals/functions.py

```diff
@@ -48,12 +50,18 @@ def concat(
     )
 
 
-def from_dict(data: dict[str, Any], *, native_namespace: Any) -> DataFrame[Any]:
+def from_dict(
+    data: dict[str, Any],
+    schema: dict[str, DType] | Schema | None = None,
+    *,
+    native_namespace: Any,
+) -> DataFrame[Any]:
     """
     Instantiate DataFrame from dictionary.
 
     Arguments:
         data: Dictionary to create DataFrame from.
+        schema: The DataFrame schema as dict of {name: type}, unused for pandas-like DataFrame's.
```

Comment:
Schema has to be complete if provided (meaning all columns should be specified, and only supports Schema, dicts or None for now. Polars has more "variety"